python/mxnet/cuda_utils.py

```diff
@@ -0,0 +1,37 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""Provides python interface to cuda-related functions of the mxnet library."""
+from .base import _LIB, mx_uint, c_str, check_call
+
+def nvtx_range_push(name, color=13323030):
```

Comment:
I don't enjoy suggesting a code repetition, but isn't the following an improvement over letting python users have to make their own colors:
```
class Nvtx:
    # Palette of colors
    RED = 0xFF0000
    GREEN = 0x00FF00
    BLUE = 0x0000FF
    YELLOW = 0xB58900
    ORANGE  = 0xCB4B16
    RED = 0xDC322F
    MAGENTA = 0xD33682
    VIOLET = 0x6C71C4
    BLUE1   = 0x268BD2
    CYAN    = 0x2AA198
    GREEN  = 0x859900

def nvtx_range_push(name, color=Nvtx.ORANGE):
```

Reply:
If you're not fond of the code repetition, at least the following smaller edit explains the magic number:
```
def nvtx_range_push(name, color=0xCB4B16):    # default color is orange
```

Reply:
I am refactoring this to be nicer and more Pythonic. Will send an updated version tomorrow, once I test it.

Reply:
The problem with this is mainly that the standard hash in Python is not deterministic, which would make colors change on subsequent runs. Using a real hashing function like the SHA from Python's hashlib on the other hand seems to be quite a heavy hammer (and most probably introduce additional overheads, which we especially do not want to do when profiling the CPU part of the training).

Reply:
What I tried to suggest was that the 'None' color would be conveyed to the backend (via -1 or alternate API with no color arg), and the backend would assign the color with the existing backend nameToColor() routine.